src/utils/localization.ts

```diff
@@ -0,0 +1,21 @@
+import { Locale, Localization } from "../types";
+
+interface LocalizationObject {
+  localization: Localization;
+}
+type LocalizedObject<T extends string> = LocalizationObject & {
+  [key in T]: string;
+};
+// interface LocalizedObject<T> extends { localization: Localization; };
+
+export const getLocalizedString = <T extends string>({
+  obj,
+  property,
+  locale,
+  fallback,
+}: {
+  obj?: LocalizedObject<T>;
```

Comment:
Should `obj` be required here? Otherwise `property` should also be optional as it's directly related to the `obj`

I was also thinking that the `fallback` should live in a centralised place rather than a string. So we'd have a file somewhere with all the fallbacks for every localised string

Reply:
The thing is fallback is going to be dependent on what you want to localize

Reply:
set property as optional in 70a6e8360b1facd340318999296cc6cdd7581ad0

Reply:
Feeling now that `obj` and `property` should be a tuple as they go hand in hand, but don't want to hold the PR due to this.